activiti-cloud-service-common/activiti-cloud-services-dbp-rest/src/main/java/org/activiti/cloud/alfresco/argument/resolver/AlfrescoPageArgumentMethodResolver.java

```diff
@@ -71,4 +81,33 @@ public Pageable resolveArgument(
             return basePageable;
         }
     }
+
+    private boolean isPaginationValueExceedinglimit(
+        AlfrescoQueryParameters alfrescoQueryParameters,
+        Pageable basePageable
+    ) {
+        if (maxItemsLimitEnabled) {
+            if (
+                alfrescoQueryParameters.getMaxItemsParameter().isSet() &&
+                alfrescoQueryParameters.getMaxItemsParameter().getValue() > maxItemsLimit
+            ) {
+                return true;
+            }
+            if (
+                alfrescoQueryParameters.getPageParameter().isSet() &&
+                alfrescoQueryParameters.getPageParameter().getValue() > maxItemsLimit
+            ) {
+                return true;
+            } else if (
```

Comment:
```suggestion
            } else if (!alfrescoQueryParameters.getMaxItemsParameter().isSet() && basePageable.getPageSize() > maxItemsLimit) {
            return true;
            }
```
I think this should work and we could remove the `return true;` at the end. WDYT?